### PR TITLE
feat: ComponentHeader component, component overview fixes, more

### PR DESCRIPTION
**Pull request by TheSisb**
## Structural changes

- [x] Swap the default theme border-color light/dark token to match Starch
- [x] Shuffle around the folder and files in the paste-website package.  Having folders of just index files is more cumbersome than just putting the file in that parent folder directly.
- [x] Removed website building from `yarn build` command, but added it to CircleCI.  When we run `yarn build` locally, we usually don't want to build the website, so this makes it faster.
- [x] Our pre-push git hook now doesn't run build anymore.  It took way too long.
- [x] Added `yarn type-check` to CircleCI and on `pre-push` git hook.
- [x] Fix some type-check errors.

## Feature changes

- [x] Add breadcrumb component for doc site
- [x] Fixed border issues on Sidebar
- [x] Keep sidebar category open when navigating to subpages
- [x] Add `ComponentHeader` shortcode component
- [x] Improve `ComponentOverviewTable` design

### CODE REVIEW

File: packages/paste-theme-tokens/src/sendgrid/index.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ const lineHeights = {
   lineHeight50: '1.75rem', // #28px
   lineHeight60: '2rem', // #32px
   lineHeight70: '2.25rem', // #36px
+  lineHeight80: '2.5rem', // #40px
```
Comment (TheSisb) [inline packages/paste-theme-tokens/src/sendgrid/index.ts]: Adding this newly needed token

File: packages/paste-website/src/components/shortcodes/ComponentHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  const {description, status, name: packageName, version} = data[0].node;
+
+  return (
+    <>
+      <ComponentHeaderBasic name={name} />
+      <Text fontSize="fontSize30" mb="space50">
+        {description}
+      </Text>
+      <Box mb="space100">
```
Comment (SiTaggart) [inline packages/paste-website/src/components/shortcodes/ComponentHeader.tsx]: Description list for the key-value pairs?

File: packages/paste-website/src/components/Input.tsx [hunk 3]
```diff
@@ -2,18 +2,16 @@ import * as React from 'react';
 import styled from '@emotion/styled';
```
Comment (SiTaggart) [inline packages/paste-website/src/components/Input.tsx]: I'm generally of the opinion that every component should eventually have some tests, even if it's just a snapshot.

Is it worth moving these to named files, given we'll create a `__test__` folder for each under it's own directory in components?

Comment (TheSisb) [reply packages/paste-website/src/components/Input.tsx]: Hmmm.. we can create sibling `Input.spec.ts` files in the same folder.  Thoughts?

Comment (SiTaggart) [reply packages/paste-website/src/components/Input.tsx]: kinda messy, and isn't the convention `__test__` folders?

File: packages/paste-website/src/components/Breadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+const CrumbSlash: React.FC<{}> = () => <CrumbSlashStyled>/</CrumbSlashStyled>;
+
+const Breadcrumb: React.FC<{}> = ({children}) => {
+  return (
+    <>
+      {React.Children.map(children, child => (
+        <>
+          {child}
+          <CrumbSlash />
```
Comment (richbachman) [inline packages/paste-website/src/components/Breadcrumb.tsx]: Can we add a check to see not show this `<CrumbSlash />` after the last breadcrumb. Not a blocker, just a thought.

File: packages/paste-website/src/components/Input.tsx (hunk 3 above)
Comment (TheSisb) [reply packages/paste-website/src/components/Input.tsx]: It isn't too bad.  Kind of looks like https://github.com/modulz/radix/tree/master/packages/radix/src/components

Or we can still have a __tests__ folder with all the component filenames repeated like
```
| __tests__
   -| Input.tsx
   -| ...
| Input.tsx
```

File: packages/paste-website/src/components/Breadcrumb.tsx (hunk 4 above)
Comment (TheSisb) [reply packages/paste-website/src/components/Breadcrumb.tsx]: This is what the design wanted /shrug

File: packages/paste-website/src/components/shortcodes/ComponentHeader.tsx (hunk 2 above)
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/ComponentHeader.tsx]: Nice, added.

File: packages/paste-website/src/components/Input.tsx (hunk 3 above)
Comment (TheSisb) [reply packages/paste-website/src/components/Input.tsx]: Reverted the changes in this PR for now.

File: packages/paste-design-tokens/formatters/sketchpalette.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ interface ColorRGB {
 }
 
 const convertToSketchPaletteColor = (input: string): {[key: string]: number} => {
+  // @ts-ignore FIXME
```
Comment (TheSisb) [inline packages/paste-design-tokens/formatters/sketchpalette.ts]: We should fix this, the typing is way off.  Didn't want it to block the PR though.

File: packages/paste-website/src/components/tokens-list/index.tsx [hunk 6]
```diff
@@ -106,8 +106,8 @@ export const TokensList: React.FC<TokensListProps> = props => {
                   <thead>
                     <Tr>
                       <Th>Token</Th>
-                      <Th width="250px">Value</Th>
-                      <Th width="250px">Example</Th>
+                      <Th style={{width: '250px'}}>Value</Th>
```
Comment (TheSisb) [inline packages/paste-website/src/components/tokens-list/index.tsx]: I realized we could revert the changes we made to the table component and use an inline-style instead.

File: packages/paste-core/utilities/text/src/index.tsx [hunk 7]
```diff
@@ -16,7 +16,6 @@ import {ThemeShape} from '@twilio-paste/theme-tokens';
 
 interface MarginProps {
   m?: keyof ThemeShape['space'];
-  margin?: keyof ThemeShape['space'];
```
Comment (SiTaggart) [inline packages/paste-core/utilities/text/src/index.tsx]: Curious why you chose to remove just `margin` from the interface?

**Comment by TheSisb** [comment]
Yeah, I totally agree.  I hadn't realized at the time that it would balloon this much.  I could still split it up if you'd like.

File: packages/paste-core/utilities/text/src/index.tsx (hunk 7 above)
Comment (TheSisb) [reply packages/paste-core/utilities/text/src/index.tsx]: I was getting a typescript error about it... but I am not anymore.  Re-added it

File: packages/paste-theme-tokens/package.json [hunk 8]
```diff
@@ -13,7 +13,7 @@
     "dist"
   ],
   "scripts": {
-    "build": "yarn clean && yarn compile",
```
Comment (SiTaggart) [inline packages/paste-theme-tokens/package.json]: This one doesn't need a clean?

Comment (TheSisb) [reply packages/paste-theme-tokens/package.json]: Good catch.  I think this is an artifact from when I was debugging the js files being built in the src folder.

File: packages/paste-website/src/components/breadcrumb/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        </>
+      ))}
+    </>
+  );
+};
+
+const BreadcrumbItem = styled(Link)`
+  height: 24px;
+  font-size: 14px;
+  line-height: 1.71;
```
Comment (SiTaggart) [inline packages/paste-website/src/components/breadcrumb/index.tsx]: Can you write a TODO here to convert these to tokens? line-heights are coming and font-size should already cover it.

Also is the height going to cause a problem if you bump the base font size for the page? Will the font scale (when using rems from tokens) and the fixed height cause issues?
